homeassistant/components/rfxtrx/__init__.py

```diff
@@ -270,7 +270,9 @@ def get_devices_from_config(config, device):
         fire_event = entity_info[ATTR_FIRE_EVENT]
         datas = {ATTR_STATE: False, ATTR_FIRE_EVENT: fire_event}
 
-        new_device = device(entity_info[ATTR_NAME], event, datas, signal_repetitions)
+        new_device = device(
+            entity_info[ATTR_NAME], event.device, datas, signal_repetitions
+        )
         RFX_DEVICES[device_id] = new_device
```

Comment:
A comment for the future: We shouldn't store the entities in a shared container. The entity is a platform concern only. That's why we use the dispatch helper to signal the entity.

It's ok to store some stable reference to the entity, eg a device id or the entity unique_id, to keep track of what entities have been created. This is often required to not create duplicate entities.

Reply:
Also, we should name `Entity` instances `entity` and not `device`.

Reply:
Done in the other pull for config flow. But it brings with it allowing overlapping entities, maybe not such a big problem if auto add is of by default.

Will see if I have the stamina to reinstate a partial step config flow. I've spent way way way too many hours of my vacation on this inyegration that I'm likely to use very little.

Reply:
The entity rename stuff is done there too.